streams/src/test/java/org/apache/kafka/streams/tests/StreamsSmokeTest.java

```diff
@@ -62,16 +65,23 @@ public static void main(final String[] args) throws InterruptedException, IOExce
                 final int numKeys = 10;
                 final int maxRecordsPerKey = 500;
                 if (disableAutoTerminate) {
-                    SmokeTestDriver.generate(kafka, numKeys, maxRecordsPerKey, false);
+                    generate(kafka, numKeys, maxRecordsPerKey, Duration.ZERO, false);
                 } else {
-                    final Map<String, Set<Integer>> allData = SmokeTestDriver.generate(kafka, numKeys, maxRecordsPerKey, true);
+                    // slow down data production to span 30 seconds so that system tests have time to
+                    // do their bounces, etc.
+                    final Map<String, Set<Integer>> allData =
+                        generate(kafka, numKeys, maxRecordsPerKey, Duration.ofSeconds(30), true);
                     SmokeTestDriver.verify(kafka, allData, maxRecordsPerKey);
                 }
                 break;
             case "process":
-                // this starts a KafkaStreams client
-                final SmokeTestClient client = new SmokeTestClient(UUID.randomUUID().toString());
-                client.start(streamsProperties);
+                // this starts the stream processing app
+                new SmokeTestClient(UUID.randomUUID().toString()).start(streamsProperties);
+                break;
+            case "process-eos":
```

Comment:
added the ability to run the smoke test app with EOS

Reply:
Can we use `StreamsEosTest` instead? See my other comment below.